### PR TITLE
fix(python): ensure `every` type is properly normalised (for `groupby_dynamic` and `groupby_rolling`)

### PR DESCRIPTION
**Pull request by alexander-beedie**
_(As raised/spotted in the python-polars [Discord](https://discord.com/channels/908022250106667068/911186243465904178/1079812872147841106))._

When calling into either of these group functions directly from LazyFrame, the `every` param was not normalised to string-form if `offset` and `period` were None, meaning `timedelta` inputs would cause the following exception:
```python
PanicException: expected a unit in the duration string
```
Fixed, and added some test coverage.

### CODE REVIEW

File: py-polars/tests/unit/operations/test_groupby.py [hunk 1]
```diff
@@ -538,7 +538,7 @@ def test_groupby_dynamic_iter() -> None:
     result2 = [
         (name, data.shape)
         for name, data in df.groupby_dynamic(
-            "datetime", every="1h", closed="left", by="a"
+            "datetime", every=timedelta(hours=1), closed="left", by="a"
```
Comment (ritchie46) [inline py-polars/tests/unit/operations/test_groupby.py:541]: Does it still work with the `"1h"` syntax?

Comment (alexander-beedie) [reply py-polars/tests/unit/operations/test_groupby.py:541]: Of course ;)  I was just extending a little bonus coverage there - a few lines above it uses "1h" in the same param.

Comment (ritchie46) [reply py-polars/tests/unit/operations/test_groupby.py]: Thanks, it wasn't clear to me from the code. Thanks a lot. :)

P.S. drinking your coffee atm, while completing out of core sort. :sunglasses: :

Comment (alexander-beedie) [reply py-polars/tests/unit/operations/test_groupby.py]: Haha, nice! I hope your coffee-loving friend approved of it too... ;)

**Pull request merged**